projects/example-app/src/app/books/reducers/index.ts

```diff
@@ -122,7 +123,9 @@ export const getSearchResults = createSelector(
   getBookEntities,
   getSearchBookIds,
   (books, searchIds) => {
-    return searchIds.map(id => books[id]);
+    return searchIds
+      .map(id => books[id])
+      .filter((book): book is Book => book != null);
```

Comment:
Any reason not to use `filter(Boolean)`?

Reply:
`.filter(Boolean)` doesn't work with `strictNullChecks`
[PLAYGROUND LINK](https://www.typescriptlang.org/play/#src=const%20arr%3A%20(number%20%7C%20undefined)%5B%5D%20%3D%20%5B1%2C%202%2C%20undefined%2C%203%2C%204%2C%20undefined%5D%3B%0D%0A%0D%0Aconst%20filtered%3A%20number%5B%5D%20%3D%20arr.filter(Boolean)%3B%0D%0A%0D%0A%0D%0A)

<img width="908" alt="Screen Shot 2019-04-13 at 3 56 03 PM" src="https://user-images.githubusercontent.com/2830407/56084689-abfee680-5e04-11e9-81ea-445306a401cc.png">